idx/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,441 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/mattbaird/elastigo/lib"
+	"github.com/raintank/met"
+	"github.com/raintank/metrictank/idx"
+	"github.com/raintank/metrictank/idx/memory"
+	"github.com/raintank/worldping-api/pkg/log"
+	"github.com/rakyll/globalconf"
+	"gopkg.in/raintank/schema.v1"
+)
+
+var (
+	idxEsOk             met.Count
+	idxEsFail           met.Count
+	idxEsAddDuration    met.Timer
+	idxEsDeleteDuration met.Timer
+
+	Enabled        bool
+	Index          string
+	Hosts          string
+	User           string
+	Pass           string
+	RetryInterval  string
+	MaxConns       int
+	MaxBufferDocs  int
+	BufferDelayMax string
+)
+
+func ConfigSetup() {
+	esIdx := flag.NewFlagSet("elasticsearch-idx", flag.ExitOnError)
+	esIdx.BoolVar(&Enabled, "enabled", false, "")
+	esIdx.StringVar(&Index, "index", "metric", "Elasticsearch index name for storing metric index.")
+	esIdx.StringVar(&Hosts, "hosts", "localhost:9200", "comma separated list of elasticsearch address in host:port form")
+	esIdx.StringVar(&User, "user", "", "HTTP basic Auth username")
+	esIdx.StringVar(&Pass, "pass", "", "HTTP basic Auth password")
+	esIdx.StringVar(&RetryInterval, "retry-interval", "1h", "Interval to retry indexing Definitions in ES after failures")
```

Comment:
in my experience, 1h is very conservative.
When we try to ingest a whole bunch of metrics all at once and there's a bunch of indexing failures, things usually settle down quickly (I think in a matter of a minute or so).  So 1h is a needlessly long wait. I would play more safe and decrease this default to a minute or two.